### DISCUSSION TITLE
Composition API: how to programmatically set global defaults?

### DISCUSSION THREAD

**Discussion by MetRonnie** [vuetifyjs/vuetify#19086] [Q&A]
How can you update the Vuetify defaults while the app is running?

In the Options API, this could be done like so
```js
this.$vuetify.defaults.global.transition = reducedAnimation ? false : null
```

There is very little documentation on the `useDefaults` composable; it looks like it cannot provide this functionality.

The best I have come up with is to wrap the application in a `VDefaultsProvider`:

```js
// App.vue
<template>
  <v-defaults-provider :defaults="vuetifyDefaults">
    <v-app>
      ...
</template>

<script setup>
...
const vuetifyDefaults = computed(() => ({
  global: {
    transition: reducedAnimation ? false : null,
    ...
</script>
```

Is there a better way?

**Comment by carvenl** [comment]
This configuration code seems to be mounted at Vue startup, and cannot be changed during the runtime.
